src/main/java/ru/hogwarts/school/service/StudentService.java

```diff
@@ -120,4 +129,63 @@ public List<String> getNamesStartingWithA() {
                 .sorted()
                 .collect(Collectors.toList());
     }
+
+    public List<Student> printParallelStudents() {
+        logger.info("Вызван метод для вывода в консоль имена студентов в параллельном режиме");
+        List<Student> students = studentRepository.findAll();
+        CheckingTheNumberOfStudents();
+
+        // Создаем пул потоков
+        ExecutorService executor = Executors.newFixedThreadPool(3);
+
+        printStudentName(students.get(FIRST_INDEX).getName());
+        printStudentName(students.get(SECOND_INDEX).getName());
+
+        executor.submit(() -> printStudentNames(students, THIRD_INDEX, FOURTH_INDEX));
+        executor.submit(() -> printStudentNames(students, FIFTH_INDEX, SIXTH_INDEX));
+        // Завершаем работу пула потоков
+        executor.shutdown();
+        return students;
+    }
+
+    // Синхронизированный метод для вывода имени студента
+    public synchronized void printStudentName(String name) {
+        System.out.println(name);
+    }
+
+    public List<Student> printSynchronizedStudents() {
+        logger.info("Вызван метод для вывода в консоль имена студентов в синхронном режиме");
+        List<Student> students = studentRepository.findAll();
+        CheckingTheNumberOfStudents();
+
+        printStudentName(students.get(FIRST_INDEX).getName());
+        printStudentName(students.get(SECOND_INDEX).getName());
+
+        // Создаем пул потоков
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+
+        executor.submit(() -> {
+            printStudentName(students.get(THIRD_INDEX).getName());
+            printStudentName(students.get(FOURTH_INDEX).getName());
+        });
+
+        executor.submit(() -> {
+            printStudentName(students.get(FIFTH_INDEX).getName());
+            printStudentName(students.get(SIXTH_INDEX).getName());
+        });
+        // Завершаем работу пула потоков
+        executor.shutdown();
+        return students;
+    }
+    private void printStudentNames(List<Student> students, int index1, int index2) {
+        System.out.println(students.get(index1).getName());
+        System.out.println(students.get(index2).getName());
+    }
+    private void CheckingTheNumberOfStudents(){
```

Comment:
Методы с маленькой буквы!
```suggestion
    private void сheckingTheNumberOfStudents(){
```